### PR TITLE
backend/feat/fare-policy-intercity-source-destination-details

### PR DESCRIPTION
**Pull request by RahulGupta0706**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: .psc-ide-port [hunk 1]
```diff
@@ -0,0 +1 @@
+15745
```
Comment (khuzema786) [inline .psc-ide-port:1]: Please revert

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Beam/FareParameters.hs [hunk 2]
```diff
@@ -46,6 +46,7 @@ data FareParametersT f = FareParametersT
     govtChargesAmount :: B.C f (Maybe HighPrecMoney),
     customerCancellationDues :: B.C f (Maybe HighPrecMoney),
     tollCharges :: B.C f (Maybe HighPrecMoney),
+    stateEntryPermitCharges :: B.C f (Maybe HighPrecMoney),
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Beam/FareParameters.hs:49]: Let's move this to Fare Parameters Intercity Details

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/FarePolicy.hs [hunk 3]
```diff
@@ -48,6 +48,7 @@ data FarePolicyD (s :: DTC.UsageSafety) = FarePolicy
     distanceUnit :: DistanceUnit,
     govtCharges :: Maybe Double,
     tollCharges :: Maybe HighPrecMoney,
+    stateEntryPermitCharges :: Maybe HighPrecMoney,
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/FarePolicy.hs:51]: Should be moved to Fare Policy Intercity Details

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/Merchant.hs [hunk 4]
```diff
@@ -1333,6 +1335,7 @@ postMerchantConfigFarePolicyUpsert merchantShortId opCity req = do
       let allowedTripDistanceBounds = Just $ FarePolicy.AllowedTripDistanceBounds {distanceUnit, minAllowedTripDistance, maxAllowedTripDistance}
       let serviceCharge :: (Maybe HighPrecMoney) = readMaybeCSVField idx row.serviceCharge "Service Charge"
       let tollCharges :: (Maybe HighPrecMoney) = readMaybeCSVField idx row.tollCharges "Toll Charge"
+      stateEntryPermitCharges :: Maybe HighPrecMoney <- readCSVField idx row.stateEntryPermitCharges "State Entry Permit Charges"
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/Merchant.hs:1338]: `readMaybeCSVField` should be used

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/FarePolicy.hs [hunk 5]
```diff
@@ -53,6 +53,7 @@ update' farePolicy = do
       Se.Set BeamFP.govtCharges $ farePolicy.govtCharges,
       Se.Set BeamFP.serviceCharge $ roundToIntegral <$> farePolicy.serviceCharge,
       Se.Set BeamFP.tollCharges $ farePolicy.tollCharges,
+      Se.Set BeamFP.stateEntryPermitCharges $ farePolicy.stateEntryPermitCharges,
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/FarePolicy.hs:56]: Migration for this column is missing for both policy and parameters tables

**Review by 0utkarsh — CHANGES_REQUESTED** [review]

**Review by khuzema786 — APPROVED** [review]

**Review by 0utkarsh — APPROVED** [review]

**Pull request merged**